client/src/components/carousel.jsx

```diff
@@ -1,12 +1,60 @@
+import Thumbnails from './thumbnails.jsx';
+import Scaled from './scaled.jsx';
+import axios from 'axios';
+import styles from './styles.css.js';
+
+
+const dummy = [ //local drive loading for development only
+  {url: './placeholder.jpg', sort: 0},
+  {url: './placeholder.jpg', sort: 1},
+  {url: './placeholder.jpg', sort: 2},
+  {url: './placeholder.jpg', sort: 3},
+  {url: './placeholder.jpg', sort: 4},
+  {url: './placeholder.jpg', sort: 5},
+  {url: './placeholder.jpg', sort: 6}
+];
+
 class Carousel extends React.Component {
   constructor(props) {
     super(props);
+    this.state = {
+      images: dummy,
+      scaled: dummy[0]
+    };
+    this.retrieveImageDocument = this.retrieveImageDocument.bind(this);
+  }
+
+  retrieveImageDocument() {
+    let endpoint = window.location.pathname + 'retrieve';
+    axios({
+      method: 'GET',
+      url: endpoint
+    })
+      .then((document) => {
```

Comment:
This is a great idea setting the state instead of just passing down the promises. I was wondering why I couldn't get mine working.